src/components/Card.js

```diff
@@ -1,37 +1,47 @@
 export default class Card {
-  constructor({ title, url }, cardSelector, handleImageClick) {
-    this._data = { title, url };
-    this._name = title;
-    this._link = url;
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClicks
+  ) {
+    this._name = name;
+    this._link = link;
+    this._id = _id;
+    this.isLiked = isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClicks = handleLikeClicks;
   }
 
   _setEventListeners() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeIcon(this);
-      });
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeClicks(this);
+    });
 
-    this._cardElement
-      .querySelector(".card__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteCard();
-      });
+    this._deleteButton.addEventListener("click", () => {
+      this._handleDeleteClick(this);
+    });
 
     this._cardImage.addEventListener("click", () => {
-      this._handleImageClick(this._data);
+      this._handleImageClick({
+        name: this._name,
+        link: this._link,
+      });
     });
   }
 
-  _handleLikeIcon() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
+  renderLikes() {
+    if (this.isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    } else {
+      this._likeButton.classList.remove("card__like-button_active");
+    }
   }
 
-  _handleDeleteCard() {
+  handleDeleteCard() {
```

Comment:
Needs correction: please rename this function to `deleteCard` as it's not an event handler anymore